File: example_files/debug-example.py
# To debug a cell, add a break point, and then hit ctrl-shift-alt-enter when in a cell.
temp_var = 0
temp_arr = []


def example_fct(i):
    global temp_var
    temp_var = i
    temp_arr.append(temp_var)


for i in range(10):
    example_fct(i)

File: example_files/debug-example.r
# Initialize variables
temp_var <- 0
temp_arr <- c()

# Define the function
example_fct <- function(i) {
  temp_var <<- i
  temp_arr <<- c(temp_arr, temp_var)
}

# Loop through 0 to 9 and call the function
for (i in 0:9) {
  example_fct(i)
}

# Print the results
temp_arr

